src/context/AirDBContext.tsx

```diff
@@ -178,61 +226,163 @@ export const AirDBProvider = React.memo((props: any) => {
         user => user.fields.username === window.localStorage.getItem('username')
       )[0] || { fields: { ID: 0 } }
     ).fields.ID
-    postAirDB('Testy - Test Scores', {
+
+    postAirDB('Test Scores', {
       userid,
       test: id
     }).then((r: any) => {
       setScoreID(r[r.length - 1].fields.ID)
     })
   }
 
-  const handleQuestionAnswer = ({
+  const answerQuestion = ({
     handleSubmit,
     value,
     question,
     index,
     scoreID
   }: any) => {
-    handleSubmit(value, question, index)
-
-    postAirDB('Testy - Test Instances', {
-      answer: value,
-      'correct answer': question.fields.answer,
-      correct: JSON.stringify(question.fields.answer === value),
-      scoreid: scoreID,
-      question: question.fields.question
+    Promise.all([
+      handleSubmit(value, question, index),
+      postAirDB('Test Instances', {
+        answer: value,
+        'correct answer': question.fields.answer,
+        correct: `${question.fields.answer === value}`,
+        scoreid: scoreID,
+        question: question.fields.question
+      })
+    ])
+  }
+
+  const sendMessage = ({ groupid, text }: any) => {
+    const userid = (
+      users.filter(
+        user => user.fields.username === window.localStorage.getItem('username')
+      )[0] || { fields: { ID: 0 } }
+    ).fields.ID
+
+    return postAirDB('Messages', {
+      groupid: `${groupid}`,
+      userid: `${userid}`,
+      text
+    })
+  }
+
+  const addCourse = ({ name }: any) => {
+    const userid = (
+      users.filter(
+        user => user.fields.username === window.localStorage.getItem('username')
+      )[0] || { id: '' }
+    ).id
+
+    return postAirDB('Courses', {
+      title: name,
+      userid: userid
+    })
+  }
+
+  const addChapter = async ({ name, course }: any) => {
+    const lessonIds: string[] = (
+      courses.filter(thisCourse => thisCourse.id === course)[0] || {
+        fields: { chapters: '' }
+      }
+    ).fields.chapters.split(', ')
```

Comment:
Not required now, but see how you might reduce the duplication between this and lines 302-306:


    const chapterIds: string[] = (
      chapters.filter(thisChapter => thisChapter.id === chapter)[0] || {
        fields: { lessons: '' }
      }
    ).fields.lessons.split(', ')


(Remember that you can access properties via key. 😉 )